sheets/api.py

```diff
@@ -755,40 +757,50 @@ def _set_spreadsheet_completed(self, file_id, completed_dt=None):
             },
         )
 
-    def fetch_incomplete_sheets(self):
+    def fetch_incomplete_sheet_metadata(self):
         """
-        Yields assignment Spreadsheets with metadata that indicate that they have not yet been completed
+        Yields assignment spreadsheet metadata for sheets that indicate that they have not yet been completed
         and should still be considered for processing.
-        NOTE: The pygsheets client includes the `open_all` method, but it loads all matching spreadsheets
-        into memory immediately, which can and will cause memory issues, hence this reimplementation.
 
         Yields:
-            pygsheets.spreadsheet.Spreadsheet: An assignment Spreadsheet that has not been completed
+            dict: An metadata dict for an assignment spreadsheet whose keys match the
+                `FILE_METADATA_FIELDS` property value
         """
         incomplete_assignment_sheet_metadata = self.expanded_sheets_client.get_metadata_for_matching_files(
             query="{} and {}".format(
                 self.ASSIGNMENT_SHEETS_QUERY, self.INCOMPLETE_SHEETS_QUERY_TERM
             ),
             file_fields=self.FILE_METADATA_FIELDS,
         )
-        for metadata_dict in incomplete_assignment_sheet_metadata:
-            yield self.pygsheets_client.open_by_key(metadata_dict["id"])
+        yield from incomplete_assignment_sheet_metadata
 
-    def get_sheet_rows(self, sheet_id):
+    def fetch_assignment_sheet(self, sheet_id):
```

Comment:
It looks like this function and `fetch_incomplete_sheet_metadata` are uncovered. Can you add a test which covers this functionality?

Reply:
See [this comment](https://github.com/mitodl/mitxpro/pull/1382/files#r356791802)